cmd/prom-aggregation-gateway/main.go

```diff
@@ -196,7 +198,20 @@ func validateFamily(f *dto.MetricFamily) error {
 	return nil
 }
 
-func (a *aggate) parseAndMerge(r io.Reader) error {
+func appendServerLabels(queryArguments url.Values, labelQueryParam string, metrics []*dto.Metric) {
+	for _, m := range metrics {
+		if _, foundLabel := queryArguments[labelQueryParam]; foundLabel {
+			for _, label := range queryArguments[labelQueryParam] {
+				l := strings.Split(label, ":")
```

Comment:
This might be more flexible as `SplitN()` ?

Reply:
Yes, if we would want to support using the separator in its value. 
Thanks, will update after https://github.com/weaveworks/prom-aggregation-gateway/pull/23#issuecomment-582939621